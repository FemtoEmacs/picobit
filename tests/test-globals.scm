(define x 2) ;; TODO the vm tries to get the car of this
(define y 3)
;; (define foo (lambda (x) (+ x 8 213 32523))) ;; TODO creating a lambda fails
;; (display (foo x))
;; (display (+ x y)) ;; actually, gives the same error as if we had a lambda

(define (foo x y)
  (if (and x y)
      (#%+ x y) ;; TODO if we use this +, it doesn't show as global, probably because it doesn't use rest params
      -1))

(foo x y)
;; duplicates have no effect on the number of constants, and 5 didn't have an effect either, might be too small
