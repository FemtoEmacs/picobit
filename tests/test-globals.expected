:0080000080
:10800000FBD700000607B582EED802209821942005
:0A80100022C1EF03EF00B87BB77543
:00000001FF
